packages/suite/src/actions/suite/metadataActions.ts

```diff
@@ -593,77 +630,80 @@ export const addAccountMetadata =
 /**
  * Generate device master-key
  * */
-export const setDeviceMetadataKey = () => async (dispatch: Dispatch, getState: GetState) => {
-    if (!getState().metadata.enabled) return;
-    const { device } = getState().suite;
-    if (!device || !device.state || !device.connected) return;
-
-    // mater key already exists
-    if (device.metadata.status === 'enabled') return;
-
-    const result = await TrezorConnect.cipherKeyValue({
-        device: {
-            path: device.path,
-            state: device.state,
-            instance: device.instance,
-        },
-        useEmptyPassphrase: device.useEmptyPassphrase,
-        path: METADATA.ENABLE_LABELING_PATH,
-        key: METADATA.ENABLE_LABELING_KEY,
-        value: METADATA.ENABLE_LABELING_VALUE,
-        encrypt: true,
-        askOnEncrypt: true,
-        askOnDecrypt: true,
-    });
+export const setDeviceMetadataKey =
+    (version: number) => async (dispatch: Dispatch, getState: GetState) => {
+        if (!getState().metadata.enabled) return;
+        const { device } = getState().suite;
+        if (!device || !device.state || !device.connected) return;
 
-    if (result.success) {
-        if (!getState().metadata.enabled) {
-            dispatch({
-                type: METADATA.ENABLE,
-            });
-        }
+        // mater key already exists
+        if (device.metadata.status === 'enabled') return;
 
-        const [stateAddress] = device.state.split('@'); // address@device_id:instance
-        const metaKey = metadataUtils.deriveMetadataKey(result.payload.value, stateAddress);
-        const fileName = metadataUtils.deriveFilename(metaKey);
-        const aesKey = metadataUtils.deriveAesKey(metaKey);
+        const isAutomatic = version > 1;
```

Comment:
nit, I'd rename `isAutomatic` to `askOnDevice`. feels like describing better what it in reality does

Reply:
it's changed in the following commit